app/src/middleware/apiTracker.js

```diff
@@ -25,13 +25,33 @@ const trackerUrls = [logUrl];
 
 // add in any token (custom or morgan built-in) we want to the format, then morgan can parse out later
 // status and response-time is a built-in morgan token
-const apiTrackerFormat = ':op :azp :ts :message :data :pattern :level :status :response-time';
+const apiTrackerFormat = ':op :azp :ts :count :message :data :pattern :level :status :response-time';
 
-const apiTrackerPattern = '%{DATA:op} %{DATA:azp} %{NUMBER:ts} %{DATA:hasMessage} %{DATA:hasData} %{DATA:hasPattern} %{DATA:hasLevel} %{NUMBER:httpStatus} %{NUMBER:responseTime}';
+const apiTrackerPattern = '%{DATA:op} %{DATA:azp} %{NUMBER:ts} %{NUMBER:logItemsCount} %{NUMBER:messageFieldCount} %{NUMBER:dataFieldCount} %{NUMBER:patternFieldCount} %{NUMBER:levelFieldCount} %{NUMBER:httpStatus} %{NUMBER:responseTime}';
 
 const stashMessage = async msg => {
+  // this is a good way of showing how we eat our own dog food and can turn a string into JSON
   const body = { message: msg.trim(), pattern: apiTrackerPattern };
-  const clogsMessage = await messageParser.parse(config.get('keycloak.clientId'), body);
+  const parsed = await messageParser.parse(config.get('keycloak.clientId'), body);
+  // but let's just massage this object a bit...
+  const clogsMessage = {...parsed};
+  // move all the counts into a sub-object of data...
+  clogsMessage.clogs.data.items = {
+    count: parsed.clogs.data.logItemsCount,
+    fields: {
+      message: parsed.clogs.data.messageFieldCount,
+      data: parsed.clogs.data.dataFieldCount,
+      pattern: parsed.clogs.data.patternFieldCount,
+      level: parsed.clogs.data.levelFieldCount
+    }
+  };
```

Comment:
With how many times we're pulling from parsed.clogs.data, may be better to have a const data = parsed.clogs.data assignment after line 35.

Reply:
👍